src/pe/import.rs

```diff
@@ -263,18 +263,29 @@ impl<'a> SyntheticImportDirectoryEntry<'a> {
             }
         };
 
-        let import_address_table_offset = &mut utils::find_offset(
-            import_directory_entry.import_address_table_rva as usize,
-            sections,
-            file_alignment,
-            opts,
-        )
-        .ok_or_else(|| {
-            error::Error::Malformed(format!(
-                "Cannot map import_address_table_rva {:#x} into offset for {}",
-                import_directory_entry.import_address_table_rva, name
-            ))
-        })?;
+        let rva = match import_directory_entry.import_address_table_rva.is_zero() {
+            true => import_directory_entry.import_lookup_table_rva,
+            false => import_directory_entry.import_address_table_rva,
+        };
```

Comment:
I am pretty sure I should check if `opts.resolve_rva` is `true` in match-false bracket (otherwise raises malformed as well as how it does before) in order to use address table which is replaced with absolute address of import address when the image is mapped.

```rs
let rva = match import_directory_entry.import_address_table_rva.is_zero() {
    true => Some(import_directory_entry.import_lookup_table_rva),
    false => opts.resolve_rva.then(|| import_directory_entry.import_address_table_rva),
};

if rva.is_none() {
    // Malformed
}
```

Reply:
the second is somewhat harder to read at first, but i think I like it better :)

Reply:
when you say you're pretty sure, you mean that this PR needs that amendment? And are you worried about some kind of regression in doing this (i.e., less strict, problems further down the parsing pipeline or how come you didn't add it to this PR?)

Reply:
If I am correct that `resolve_rva` opt is meant to be _solely_ used for the mapped image (i.e., raw memory dumps, minidumps etc):

Yes, but what I am not sure is that we already make use of address table right now which is rewritten to the 64/32-bit absolute address when mapped into the memory. So If this is not to be the problem for this long in goblin, then I see no problems there to use lookup table.

I guess this is some kind of non-well implementation.. we should look for lookup tables rather than address tables.

- AddressTable/FTs: may not be valid in memory dumps.
  - Unlike PLT/GOT in ELF, In x86-64 arch and PE64, rel32 RVA of import calls like `call __imp_MessageBoxA` are actually points to the FTs.
- LookupTable/OFTs: always correct regardless of mapped images (i.e., for dependency walks)

I came to the conclusion that we do not need this optimization at this time.